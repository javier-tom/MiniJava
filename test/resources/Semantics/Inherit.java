class Main {
    public static void main(String[] args) {
        System.out.println(new A().run());
    }
}

class B {
    public int run() {
        return 17;
    }
}

class A extends B {
    public int whatever(int stuff) {
        return stuff;
    }
}

