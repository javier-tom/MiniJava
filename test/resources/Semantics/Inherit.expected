class A:
  super: B
  fields:
  method run:
    params:
    returns: int
    locals:
  method whatever:
    params: int,
    returns: int
    locals:

class B:
  super: null
  fields:
  method run:
    params:
    returns: int
    locals:
